lib/src/rules/invariant_booleans.dart

```diff
@@ -182,9 +182,10 @@ class _ContradictionReportRule extends LintRule {
 }
 
 TestedExpressions _findPreviousTestedExpressions(BinaryExpression node) {
-  Block block = node
-      .getAncestor((a) => a is Block && a.parent is BlockFunctionBody);
-  Iterable<AstNode> nodesInDFS = DartTypeUtilities.traverseNodesInDFS(block);
+  Block block =
+      node.getAncestor((a) => a is Block && a.parent is BlockFunctionBody);
```

Comment:
I'm guessing that this should just be `FunctionBody`, but I haven't tested it. Consider:
```
someFunction() {
  int a = new DateTime.now().millisecondsSinceEpoch;
  innerFunction() => a > 0 ? false : null;
  if (a > 0) { // OK
    print('bla');
  }
}
```
If that doesn't currently fail, then there's no need to change the test.